NewClient panics with non-Transport RoundTripper

* Vault Version:
* Operating System/Architecture: current HEAD, `6775fe4927ccc99b390d2b32eb4350878f827ed5`

**Expected Behavior:**
I should be able to construct a Go client instance with an `http.Client` value whose `Transport` is a `http.RoundTripper` that is not a `*http.Transport`.

**Actual Behavior:**
`NewClient` panics

**Steps to Reproduce:**
Add the following to `api/client_test.go` and run `go test`:
```
type roundTripperFunc func(*http.Request) (*http.Response, error)

func (rt roundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
	fmt.Println("begin trace")
	defer fmt.Println("end trace")
	return rt(r)
}

func TestClientInstrumentingRoundTripper(t *testing.T) {
	client := &http.Client{
		Transport: roundTripperFunc(http.DefaultTransport.RoundTrip),
	}

	_, err := NewClient(&Config{
		HttpClient: client,
	})
	if err != nil {
		t.Fatal(err)
	}
}
```
you'll see
```
--- FAIL: TestClientNonTransportRoundTripper (0.00s)
panic: interface conversion: http.RoundTripper is api.roundTripperFunc, not *http.Transport [recovered]
        panic: interface conversion: http.RoundTripper is api.roundTripperFunc, not *http.Transport

goroutine 64 [running]:
testing.tRunner.func1(0xc42027a0f0)
        /usr/local/go/src/testing/testing.go:711 +0x2d2
panic(0x19f64c0, 0xc420566140)
        /usr/local/go/src/runtime/panic.go:491 +0x283
github.com/hashicorp/vault/api.NewClient(0xc420566100, 0xc420566100, 0xc400000008, 0xc420059a80)
        /Users/bhcleek/go/src/github.com/hashicorp/vault/api/client.go:274 +0x3e8
github.com/hashicorp/vault/api.TestClientNonTransportRoundTripper(0xc42027a0f0)
        /Users/bhcleek/go/src/github.com/hashicorp/vault/api/client_test.go:212 +0xea
testing.tRunner(0xc42027a0f0, 0x1b69e30)
        /usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:789 +0x2de
exit status 2
FAIL    github.com/hashicorp/vault/api  0.247s
```

**Important Factoids:**
This is because `NewClient` asserts that the `HTTPClient.Transport` is a `*http.Transport`. This assertion is often not valid, and effectively prevents any number of instrumenters from being used.

--- a/api/client.go
+++ b/api/client.go
@@ -271,9 +271,10 @@ func NewClient(c *Config) (*Client, error) {
 		c.HttpClient.Transport = cleanhttp.DefaultTransport()
 	}
 
-	tp := c.HttpClient.Transport.(*http.Transport)
-	if err := http2.ConfigureTransport(tp); err != nil {
-		return nil, err
+	if tp, ok := c.HttpClient.Transport.(*http.Transport); ok {
+		if err := http2.ConfigureTransport(tp); err != nil {
+			return nil, err
+		}
 	}
 
 	redirFunc := func() {

--- a/api/client_test.go
+++ b/api/client_test.go
@@ -194,3 +194,22 @@ func TestClientTimeoutSetting(t *testing.T) {
 	}
 
 }
+
+type roundTripperFunc func(*http.Request) (*http.Response, error)
+
+func (rt roundTripperFunc) RoundTrip(r *http.Request) (*http.Response, error) {
+	return rt(r)
+}
+
+func TestClientNonTransportRoundTripper(t *testing.T) {
+	client := &http.Client{
+		Transport: roundTripperFunc(http.DefaultTransport.RoundTrip),
+	}
+
+	_, err := NewClient(&Config{
+		HttpClient: client,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}

